src/dbus/dbus_thread_object.cpp

```diff
@@ -0,0 +1,558 @@
+    RegisterGetPropertyHandler(OTBR_DBUS_PREFIX, OTBR_DBUS_NETWORK_NAME_PROPERTY,
+                               std::bind(&DBusThreadObject::GetNetworkNameHandler, this, _1));
+
+    RegisterGetPropertyHandler(OTBR_DBUS_PREFIX, OTBR_DBUS_PANID_PROPERTY,
+                               std::bind(&DBusThreadObject::GetPanIdHandler, this, _1));
+    RegisterGetPropertyHandler(OTBR_DBUS_PREFIX, OTBR_DBUS_EXTPANID_PROPERTY,
+                               std::bind(&DBusThreadObject::GetExtPanIdHandler, this, _1));
+    RegisterGetPropertyHandler(OTBR_DBUS_PREFIX, OTBR_DBUS_CHANNEL_PROPERTY,
+                               std::bind(&DBusThreadObject::GetChannelHandler, this, _1));
+    RegisterGetPropertyHandler(OTBR_DBUS_PREFIX, OTBR_DBUS_MASTER_KEY_PROPERTY,
+                               std::bind(&DBusThreadObject::GetMasterKeyHandler, this, _1));
+    RegisterGetPropertyHandler(OTBR_DBUS_PREFIX, OTBR_DBUS_CCA_FAILURE_RATE_PROPERTY,
+                               std::bind(&DBusThreadObject::GetCcaFailureRateHandler, this, _1));
+    RegisterGetPropertyHandler(OTBR_DBUS_PREFIX, OTBR_DBUS_LINK_COUNTERS_PROPERTY,
+                               std::bind(&DBusThreadObject::GetLinkCountersHandler, this, _1));
+    RegisterGetPropertyHandler(OTBR_DBUS_PREFIX, OTBR_DBUS_IP6_COUNTERS_PROPERTY,
+                               std::bind(&DBusThreadObject::GetIp6CountersHandler, this, _1));
+    RegisterGetPropertyHandler(OTBR_DBUS_PREFIX, OTBR_DBUS_SUPPORTED_CHANNEL_MASK_PROPERTY,
+                               std::bind(&DBusThreadObject::GetSupportedChannelMaskHandler, this, _1));
+
+    return err;
+}
+
+void DBusThreadObject::DeviceRoleHandler(otDeviceRole aDeviceRole)
+{
+    SignalPropertyChanged(OTBR_DBUS_PREFIX, OTBR_DBUS_DEVICE_ROLE_PROPERTY, GetDeviceRoleName(aDeviceRole));
+}
+
+void DBusThreadObject::ScanHandler(DBusRequest &aRequest)
+{
+    auto threadHelper = mNcp->GetThreadHelper();
+    threadHelper->Scan(std::bind(&DBusThreadObject::ReplyScanResult, this, aRequest, _1, _2));
+}
+
+void DBusThreadObject::ReplyScanResult(DBusRequest &                          aRequest,
+                                       otError                                aError,
+                                       const std::vector<otActiveScanResult> &aResult)
+{
+    std::vector<otbrActiveScanResult> results;
+
+    if (aError != OT_ERROR_NONE)
+    {
+        aRequest.ReplyOtbrResult(static_cast<otbrError>(aError));
+    }
+    else
+    {
+        for (const auto &r : aResult)
+        {
+            otbrActiveScanResult result;
+
+            memcpy(&result.mExtAddress, r.mExtAddress.m8, sizeof(result.mExtAddress));
+            memcpy(&result.mExtendedPanId, r.mExtendedPanId.m8, sizeof(result.mExtendedPanId));
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+            result.mExtAddress    = bswap_64(result.mExtAddress);
+            result.mExtendedPanId = bswap_64(result.mExtendedPanId);
+#endif
```

Comment:
This is not elegant to me. We need to write two separate expressions plus a conditional preprocessor to make it right, and we need to do this everywhere if we want to convert network bytes to integers. I consider it bad because:
1. it is error-prone to copy bytes directly to integers;
2. It is better to write endianness-independent code;

 I suggest providing template functions like below:
```c++
// Decode Integers
template<typename Int>
Int Decode(const uint8_t *aData, size_t aLength) {
    Int ret = 0;
    assert(aLength == sizeof(Int));
    for (size_t i = 0; i <= aLength; ++i) {
        ret = (ret << sizeof(uint8_t)) + aData[i];
    }
    return ret;
}

// Encode Integers
// ...
```

Reply:
The data structure for uint64 in OpenThread is an array which is always stored in big-endian order. So these values have to be specially handled.

Reply:
@gjc13 I understand that this is necessary. My point is to make it a function. BTW, `__BYTE_ORDER__` is not defined by the C++ specification, right?